pyrsistent/_checked_types.py

```diff
@@ -45,18 +47,50 @@ def __str__(self):
             missing_fields=', '.join(self.missing_fields))
 
 
+_preserved_iterables = (
+        Enum,
+)
+"""For types that are themselves iterable, we would mistakenly take their
+elements for the type annotation. This set defines types to 'preserve' by not
+iterating over them."""
+
+
+def maybe_type_to_list(t):
+    """Try to coerce a user-supplied type directive into a list of types."""
+    # TODO FIXME: Move type validation inside this function to consolidate it.
+    # Preserve type annotations that are thmeselves iterable.
+    if isinstance(t, type) and issubclass(t, _preserved_iterables):
+        return [t]
+    # Preserve type annotations given as strings.
+    elif isinstance(t, string_types):
+        return [t]
+    else:
+        if not isinstance(t, Iterable):
+            return [t]
+        else:
+            return list(t)
+
+
+def maybe_types_to_list(ts):
+    res = []
+
+    for t in ts:
+        res.extend(maybe_type_to_list(t))
+
+    return tuple(res)
+
+
 def _store_types(dct, bases, destination_name, source_name):
-    def to_list(elem):
-        if not isinstance(elem, Iterable) or isinstance(elem, six.string_types):
-            return [elem]
-        return list(elem)
-
-    dct[destination_name] = to_list(dct[source_name]) if source_name in dct else []
-    dct[destination_name] += sum([to_list(b.__dict__[source_name]) for b in bases if source_name in b.__dict__], [])
-    dct[destination_name] = tuple(dct[destination_name])
-    if not all(isinstance(t, type) or isinstance(t, six.string_types) for t in dct[destination_name]):
+    maybe_types = maybe_types_to_list([
```

Comment:
I'd like to cover this directly somewhere, let me know if you have a suggested locatoin to put the coverage.

Reply:
What do you mean by "covering it directly"? That you want to write a unit test for this function specifically?

In that case I would advise against it and instead exercise the functionality through the public interface of pyrstistent. It should hopefully not be too difficult and I think it gives greater flexibilities for the type of refactorings that you've just done without having to rewrite the test suite.

Reply:
yeah I wanted to unittest the internals. I now understand that y'all prefer to test through the public interface, which is fine, so this comment is obviated.